src/ant/include/ant/AntennaChecker.hh

```diff
@@ -68,6 +68,25 @@ struct Violation
   int diode_count_per_gate;
 };
 
+// A struct that contains a net as well as one of its routing levels.
+// This struct is used to index the map, allowed_wire_length which is created 
+// in the class below.
+struct NetLayerPair
+{
+    dbNet* net;
+    int routing_level;
+ 
+    // constructor
+    NetLayerPair(dbNet* net, int routing_level): net(net), routing_level(routing_level) {}
+
+    // overload `<` operator to use this struct as a key for an std::map
+    bool operator<(const NetLayerPair &ob) const {
+        return net->getId() < ob.net->getId() ||  ((ob.net->getId()  ==  net->getId()) && routing_level < ob.routing_level);
```

Comment:
This is correct but I tend to use

  return std::tie(net->getId(), routing_level) < std::tie(ob.net->getId(), ob.routing_level);

which easily generalizes to many fields.